### ISSUE TITLE
Parse annotations

### ISSUE THREAD

**Issue by thebsdbox** [plunder-app/kube-vip#153]
**Is your feature request related to a problem? Please describe.**
Not currently, I'd like to have the daemonset parse annotations if they exist

**Describe the solution you'd like**

```
Annotations:        kubeadm.alpha.kubernetes.io/cri-socket: /var/run/dockershim.sock
                    node.alpha.kubernetes.io/ttl: 0
                    packet.com/node-asn: 65000
                    packet.com/peer-asn: 65530
                    packet.com/peer-ip: 10.80.96.24
                    packet.com/src-ip: 10.80.96.25
```

Read this configuration from the node.

**Describe alternatives you've considered**

**Additional context**

**Comment by thebsdbox** [comment]
@gianarb is this something you'd be interested to look at? we'd probably need to keep `kube-vip` in a waiting state until it can read these annotations for BGP. Or have kube-vip exit.. at least then people will investigate why it isn't starting.

**Comment by gianarb** [comment]
Sure thing @thebsdbox ! Let's assume I don't know anything about this issue and I don't have any context about what you are doing.

Who sets those annotations in the cloud control manager? Are those annotations something kube-vip should wait for before doing any work? If so why?

Thanks

**Comment by thebsdbox** [comment]
Yeah. I guess my idea is a flag that is --annotations "vendor" and there is something that looks for `fmt.Sprintf("%s/peerIP", vendor)` or something like that.

**Comment by gianarb** [comment]
I have to admit I am not sure what you want and why but if you look at this:

> Yeah. I guess my idea is a flag that is --annotations "vendor" and there is something that looks for fmt.Sprintf("%s/peerIP", vendor) or something like that.

You are forcing cloud providers probably in the CCM layer to implement a set of annotations `peerIP`, `asn`, `src` and so on, each of them with their own `vendor` identifier, even if the annotation are all the same.

Why not having `kube-vip/node-asn` and so on? In this way, kube-vip stays vendor-agnostic (looking at your example I can't tell why vendor matters other than for branding)

**Comment by thebsdbox** [comment]
I was following the notation definition as set per the packet CCM? The idea of having a `--annotations "packet.com"` was just incase another CCM opted to advertise BGP configuration and could do it in the same way as the packet CCM.

**Comment by gianarb** [comment]
Ok great

**Comment by thebsdbox** [comment]
Initial use of annotations merged in #157 !!
